realsense2_camera/src/fix_set_exposure.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ *  ______                   __  __              __
+ * /\  _  \           __    /\ \/\ \            /\ \__
+ * \ \ \L\ \  __  __ /\_\   \_\ \ \ \____    ___\ \ ,_\   ____
+ *  \ \  __ \/\ \/\ \\/\ \  /'_` \ \ '__`\  / __`\ \ \/  /',__\
+ *   \ \ \/\ \ \ \_/ |\ \ \/\ \L\ \ \ \L\ \/\ \L\ \ \ \_/\__, `\
+ *    \ \_\ \_\ \___/  \ \_\ \___,_\ \_,__/\ \____/\ \__\/\____/
+ *     \/_/\/_/\/__/    \/_/\/__,_ /\/___/  \/___/  \/__/\/___/
+ */
+/**
+ * @license Apache 2.0. See LICENSE file in root directory.
+ * @copyright Copyright 2020, Avidbots Corp.
+ * @file      fix_set_autoexposure.cpp
+ * @brief     Brief description of file.
+ * @author    Paul Belanger
+ */
+
+// most of this code is adapted from the fix_set_autoexposure script in the
+// avidbots robot repo.
+
+#include <realsense2_camera/fix_set_exposure.h>
+#include <ros/console.h>
+#include <boost/interprocess/sync/named_mutex.hpp>
+
+namespace realsense2_camera
+{
+
+
+
+// through experimentation, it takes the D435 sensor about 2 seconds to
+// finish its hardware reset.
+// unfortunately there isn't a way to know for sure that a device has finished resetting.
+// if we wait for less than this time, then we'll end up picking up the original
+// sensor before it finishes disconnecting from the USB bus, leading to all sorts of
+// issues down the line when we try to pull frames.
+const static double MIN_REACQUIRE_WAIT_TIME = 2.5; //[s]
+
+/**
+ * @brief Attempt to toggle the autoexposure value of the device.
+ * @param device  The device to operate on
+ * @return  True if the setting was applied successfully, otherwise false.
+ */
+static bool try_set_autoexposure(rs2::device& device)
+{
+  const auto option = rs2_option::RS2_OPTION_ENABLE_AUTO_EXPOSURE;
+
+  rs2::sensor stereo_module;
+  bool depth_sensor_found = false;
+  for(auto& sen: device.query_sensors())
+  {
+    if(sen.is<rs2::depth_sensor>())
+    {
+      depth_sensor_found = true;
+      stereo_module = sen;
+    }
+  }
+  if(! depth_sensor_found)
+  {
+    ROS_ERROR("Could not find a depth sensor on the device! ");
+    return false;
+  }
+
+  try
+  {
+    const float original_value = stereo_module.get_option(option);
+    const float expected_new_value = (original_value > 0.0f ? 0.0f : 1.0f);
+    ROS_DEBUG_STREAM("Set autoexposure " << original_value << " -> " << expected_new_value);
+    stereo_module.set_option(option, expected_new_value);
+    const float actual_new_value = stereo_module.get_option(option);
+    ROS_DEBUG_STREAM("Actual autoexposure value: " << actual_new_value);
+    return (actual_new_value == expected_new_value);
```

Comment:
Comparing two floating points with `==`, perhaps you could convert them values into int before comparsion?

Reply:
Nvm with this one just realized the float values are not used for computation anywhere.